app/land-action/index.js

```diff
@@ -15,9 +15,46 @@ const getActions = () => {
   return actions
 }
 
+function addRule (action, newRule) {
+  if (!action.eligibilityRules) {
+    action.eligibilityRules = []
+  }
+  if (!action.eligibilityRules.some(rule => rule.id === newRule.id)) {
+    action.eligibilityRules.push(newRule)
+  }
+  const actionIndex = actions.findIndex(a => a.code === action.code)
+  if (actionIndex !== -1) {
+    actions[actionIndex] = action
+  }
+}
+
+const findRuleIndex = (eligibilityRules, ruleId) => {
+  return eligibilityRules.findIndex(rule => rule.id === ruleId)
+}
+
+const updateRule = (action, ruleIndex, newRule) => {
+  action.eligibilityRules[ruleIndex] = newRule
+  const actionIndex = actions.findIndex(a => a.code === action.code)
+  if (actionIndex !== -1) {
+    actions[actionIndex] = action
+  }
+}
+
+function deleteRule (action, ruleIndex) {
+  action.eligibilityRules.splice(ruleIndex, 1)
+  const actionIndex = actions.findIndex(a => a.code === action.code)
```

Comment:
Given that line 44 should delete the rule from the array, are lines 45-48 needed ?

It would also be worth checking the return value of the `splice` function to ensure it is an array containing the deleted element.

Reply:
> Given that line 44 should delete the rule from the array, are lines 45-48 needed ?
 Line 44 just updates a local copy of the action by deleting it's rule.
 45-48  update the in memory array of actions with the modified copy.